solc/CommandLineInterface.cpp

```diff
@@ -939,27 +924,21 @@ string CommandLineInterface::objectWithLinkRefsHex(evmasm::LinkerObject const& _
 	return out;
 }
 
-bool CommandLineInterface::assemble(
-	yul::AssemblyStack::Language _language,
-	yul::AssemblyStack::Machine _targetMachine,
-	bool _optimize,
-	optional<unsigned int> _expectedExecutionsPerDeployment,
-	optional<string> _yulOptimiserSteps
-)
+bool CommandLineInterface::assemble(yul::AssemblyStack::Language _language, yul::AssemblyStack::Machine _targetMachine)
 {
-	solAssert(_optimize || !_yulOptimiserSteps.has_value(), "");
-
 	bool successful = true;
 	map<string, yul::AssemblyStack> assemblyStacks;
 	for (auto const& src: m_fileReader.sourceCodes())
 	{
-		OptimiserSettings settings = _optimize ? OptimiserSettings::full() : OptimiserSettings::minimal();
```

Comment:
I only now noticed that `assemble()` and `compile()` are using different optimizer settings. Do we actually want `full()` here?

It does not matter all that much right now because `full()` and `standard()` are identical but they could diverge in the future.

Reply:
@chriseth

Reply:
I think by default we should use `standard()` shouldn't we?

Reply:
ok. changed to `standard()`.

I checked now and looks like in Standard JSON we just use `standard()` in both cases so it looks more like a bug than something intentional.